enrollment-server-onboarding-provider-innovatrics/src/main/java/com/wultra/app/onboardingserver/provider/innovatrics/InnovatricsDocumentVerificationProvider.java

```diff
@@ -17,35 +17,101 @@
  */
 package com.wultra.app.onboardingserver.provider.innovatrics;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+import com.wultra.app.enrollmentserver.model.enumeration.DocumentType;
+import com.wultra.app.enrollmentserver.model.enumeration.DocumentVerificationStatus;
 import com.wultra.app.enrollmentserver.model.integration.*;
+import com.wultra.app.enrollmentserver.model.integration.Image;
 import com.wultra.app.onboardingserver.api.errorhandling.DocumentVerificationException;
 import com.wultra.app.onboardingserver.api.provider.DocumentVerificationProvider;
 import com.wultra.app.onboardingserver.common.database.entity.DocumentResultEntity;
 import com.wultra.app.onboardingserver.common.database.entity.DocumentVerificationEntity;
 import com.wultra.app.onboardingserver.common.errorhandling.RemoteCommunicationException;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import com.wultra.app.onboardingserver.provider.innovatrics.model.api.*;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
 import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
 
-import java.util.List;
-import java.util.Map;
+import java.util.*;
+import java.util.stream.Collectors;
 
 /**
  * Implementation of the {@link DocumentVerificationProvider} with <a href="https://www.innovatrics.com/">Innovatrics</a>.
  *
  * @author Jan Pesek, jan.pesek@wultra.com
  */
-@ConditionalOnProperty(value = "enrollment-server-onboarding.document-verification.provider", havingValue = "innovatrics")
+@ConditionalOnExpression("""
+        '${enrollment-server-onboarding.presence-check.provider}' == 'innovatrics' and '${enrollment-server-onboarding.document-verification.provider}' == 'innovatrics'
+        """)
 @Component
-class InnovatricsDocumentVerificationProvider implements DocumentVerificationProvider {
+@AllArgsConstructor
+@Slf4j
+public class InnovatricsDocumentVerificationProvider implements DocumentVerificationProvider {
+
+    private static final Set<String> CRUCIAL_ATTRIBUTES = Set.of("documentNumber", "dateOfIssue", "dateOfExpiry", "surname", "dateOfBirth", "personalNumber", "givenNames");
```

Comment:
🤔 So far so good, but probably we may extract that to the configuration properties.